API/utils/Model.php

```diff
@@ -811,6 +811,121 @@ function RemoveAdmin($email) {
         $this->FreeStmt($stmt);
     }
 
+
+    /**
+     * @param int $projectId
+     * @param string $androidId
+     * @param string $term
+     * @param string $textBackTranslation
+     * @param string $audioRecordingFilename
+     * @param string $audioData
+     * @return int
+     */
+    public function CreateWordLinkRecording($projectId, $androidId, $term, $textBackTranslation, $audioRecordingFilename, $audioData) {
+        try {
+            $this->conn->beginTransaction();
+
+            $generatedFileName = $this->GenerateRecordingFileName($audioRecordingFilename);
+
+            // create record entry
+            $sql = "INSERT INTO WordLinkRecordings(term, projectId, textBackTranslation, fileName)
+				VALUES (?, ?, ?, ?);";
+            $stmt = $this->PrepareAndExecute($sql, array($term, $projectId, $textBackTranslation, $generatedFileName));
+            $this->FreeStmt($stmt);
+
+            $recordingId = $this->conn->lastInsertId();
+            $this->conn->commit();
+
+            # save recording file content
```

Comment:
Please save recording before commit. In case of error, orphaned files are preferable to missing files.

Reply:
During update, let's do 1. save new file, 2. commit, 3. delete old file, so that we minimize the chance of missing files.